internal/provider/resources/resource_team_roles.go

```diff
@@ -0,0 +1,329 @@
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	tflog.Trace(ctx, fmt.Sprintf("read a team_roles resource: %v", teamId))
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *teamRolesResource) Update(
+	ctx context.Context,
+	req resource.UpdateRequest,
+	resp *resource.UpdateResponse,
+) {
+	var data models.TeamRoles
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	diags := r.MutateRoles(ctx, &data)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *teamRolesResource) Delete(
+	ctx context.Context,
+	req resource.DeleteRequest,
+	resp *resource.DeleteResponse,
+) {
+	var data models.TeamRoles
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// delete request
+	teamId := data.TeamId.ValueString()
+
+	// update request with no workspace roles, no deployment roles and lowest organization role
+	updateTeamRolesRequest := iam.UpdateTeamRolesJSONRequestBody{
+		DeploymentRoles:  nil,
+		OrganizationRole: iam.UpdateTeamRolesRequestOrganizationRole(iam.ORGANIZATIONMEMBER),
+		WorkspaceRoles:   nil,
+	}
+	teamRoles, err := r.iamClient.UpdateTeamRolesWithResponse(
+		ctx,
+		r.organizationId,
+		teamId,
+		updateTeamRolesRequest,
+	)
+	if err != nil {
+		tflog.Error(ctx, "failed to delete team_roles", map[string]interface{}{"error": err})
+		resp.Diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to delete team_roles, got error: %s", err),
+		)
+		return
+	}
+	_, diagnostic := clients.NormalizeAPIError(ctx, teamRoles.HTTPResponse, teamRoles.Body)
+	if diagnostic != nil {
+		resp.Diagnostics.Append(diagnostic)
+		return
+	}
+
+	diags := data.ReadFromResponse(ctx, teamId, teamRoles.JSON200)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	tflog.Trace(ctx, fmt.Sprintf("deleted a team_roles resource for team '%v'", teamId))
+}
+
+func (r *teamRolesResource) ImportState(
+	ctx context.Context,
+	req resource.ImportStateRequest,
+	resp *resource.ImportStateResponse,
+) {
+	resource.ImportStatePassthroughID(ctx, path.Root("team_id"), req, resp)
+}
+
+// RequestWorkspaceRoles converts a Terraform set to a list of iam.WorkspaceRole to be used in create and update requests
+func RequestWorkspaceRoles(ctx context.Context, workspaceRolesObjSet types.Set) ([]iam.WorkspaceRole, diag.Diagnostics) {
+	if len(workspaceRolesObjSet.Elements()) == 0 {
+		return []iam.WorkspaceRole{}, nil
+	}
+
+	var roles []models.WorkspaceRole
+	diags := workspaceRolesObjSet.ElementsAs(ctx, &roles, false)
+	if diags.HasError() {
+		return nil, diags
+	}
```

Comment:
does this log errors if it fails to set workspaceRoles?

Reply:
yeah